libsolidity/analysis/TypeChecker.cpp

```diff
@@ -387,7 +387,7 @@ bool TypeChecker::visit(FunctionDefinition const& _function)
 			_var.referenceLocation() == VariableDeclaration::Location::Storage &&
 			!m_currentContract->abstract()
 		)
-			m_errorReporter.typeError(
+			m_errorReporter.fatalTypeError(
```

Comment:
Can you add a test for this case as well? It needs the internal type to be the parameter of a constructor.

Reply:
Added a test case in the file `calldata_struct_array_argument_with_internal_data_type_inside_as_constructor_parameter.sol`